moped-editor/src/views/projects/projectView/ProjectComponents/EditAttributesModal.js

```diff
@@ -89,7 +93,15 @@ const EditComponentModal = ({
           signals: [signalToInsert],
         },
       })
-        .then(() => onComponentSaveSuccess(updatedClickedComponentState))
+        .then(() => {
+          onComponentSaveSuccess(updatedClickedComponentState);
+          // Zoom to the new or existing signal
+          zoomMapToFeatureCollection(
+            mapRef,
+            { type: "FeatureCollection", features: [signalFromForm] },
+            fitBoundsOptions.zoomToClickedComponent
+          );
+        })
```

Comment:
i was going to ask if there was a race condition here with the `onComponentSaveSuccess` refetch, but you're zooming straight to the map data we have in memory, which isn't changing. this is snappy!

Reply:
🙏 Laziness wins 😆